tests/models.py

```diff
@@ -1352,3 +1353,89 @@ def test_operator_clear(self):
         self.assertEqual(ti2.try_number, 1)
         # try_number (0) + retries(1)
         self.assertEqual(ti2.max_tries, 1)
+
+    def test_xcom_disable_pickle_type(self):
```

Comment:
There won't be automatic switching (it would be quite hacky to add that kind of logic), but it's OK because users need to do manually change the configuration anyways.